app/lib/shared/tool_env.dart

```diff
@@ -28,17 +32,32 @@ class ToolEnvRef {
 
   Future release() async {
     _active--;
-    if (_started >= _maxCount && _active == 0) {
-      await _pubCacheDir.delete(recursive: true);
+    if (_active == 0) {
+      // Delete directory if the instance is no longer active or it reached the
+      // maximum threshold.
+      if (_started >= _maxCount || _current != this) {
+        await _pubCacheDir.delete(recursive: true);
+      }
     }
   }
 }
 
-Future<ToolEnvRef> createToolEnvRef() async {
+/// Gets a currently available [ToolEnvRef] if it is used less than the
+/// configured threshold (_maxCount, currently 100). If it it has already
```

Comment:
Don't document the current value, refer to the constant so it doesn't go out of date.